chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/curesStatistics/email/CuresStatisticsChartSpreadsheet.java

```diff
@@ -0,0 +1,299 @@
+        criteriaToRowMaps.add(new CriteraToRowMap(CertificationCriterionService.Criteria2015.F_5_CURES, F_5_CURES_ROW_IDX));
+        criteriaToRowMaps.add(new CriteraToRowMap(CertificationCriterionService.Criteria2015.G_6_CURES, G_6_CURES_ROW_IDX));
+        criteriaToRowMaps.add(new CriteraToRowMap(CertificationCriterionService.Criteria2015.G_9_CURES, G_9_CURES_ROW_IDX));
+        criteriaToRowMaps.add(new CriteraToRowMap(CertificationCriterionService.Criteria2015.G_10, G_10_ROW_IDX));
+    }
+
+    public File generateSpreadsheet(Map<CertificationCriterionDTO, CuresCriterionChartStatistic> dataMap, LocalDate reportDataDate) throws IOException {
+        File newFile = copyTemplateFileToTemporaryFile();
+        Workbook workbook = getWorkbook(newFile);
+
+        populateDataSheet(dataMap, workbook);
+
+        updateChartTitles(workbook, reportDataDate);
+
+        XSSFFormulaEvaluator.evaluateAllFormulaCells(workbook);
+
+        return writeFileToDisk(workbook, newFile);
+    }
+
+    private File writeFileToDisk(Workbook workbook, File saveFile) throws IOException {
+        try (FileOutputStream outputStream = new FileOutputStream(saveFile)) {
+            workbook.write(outputStream);
+            workbook.close();
+        }
+        return saveFile;
+    }
+
+    private void updateChartTitles(Workbook workbook, LocalDate reportDate) {
+        List<Sheet> allChartSheets = new ArrayList<Sheet>();
+        allChartSheets.add(workbook.getSheetAt(2));
+        allChartSheets.add(workbook.getSheetAt(3));
+        allChartSheets.add(workbook.getSheetAt(4));
+
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMMM d, yyyy");
+
+        for (Sheet currentSheet : allChartSheets) {
+            XSSFDrawing drawing = (XSSFDrawing) currentSheet.createDrawingPatriarch();
+            for (XSSFChart chart : drawing.getCharts()) {
+                // This goes into the XML that makes up the chart to set the data in the title.  This has potential
+                // to vary from chart to chart, based on formatting.
+                chart.getCTChart().getTitle().getTx().getRich().getPArray(1).getRArray(0).setT(reportDate.format(formatter));
+            }
+        }
+    }
+
+    private void populateDataSheet(Map<CertificationCriterionDTO, CuresCriterionChartStatistic> dataMap, Workbook workbook) {
+        Sheet sheet = getDataSheet(workbook);
+
+        criteriaToRowMaps.stream()
+                .forEach(map ->
+                        writeDataForCuresCriterionChartStatistic(
+                                getCuresCriterionChartStatisticByCriterion(
+                                        dataMap, criterionService.get(map.getCriteriaKey())), sheet.getRow(map.getRowNumber())));
+
+        Sheet sortedDataSheet = getDataSortedSheet(workbook);
+        CopyAndSortWorksheet.copy(sheet, sortedDataSheet, PERCENT_CURES_COL_IDX, false);
+    }
+
+
+
```

Comment:
I've noticed a few places where there are a bunch of blank lines in some of these files. Is that intentional? Is there a particular reason for these? I wouldn't hold up approval on stuff like this, but it doesn't seem to be consistent and inconsistent things call out to me, which distracts me from reading the actual code